### ISSUE TITLE
iCloud3 v3.0.5.1: No update of FamShr-Devices

### ISSUE THREAD

**Issue by Sentinel79** [gcobb321/icloud3_v3#305]
Hi,

after updating to iCloud3 v3.0.5.1, I have the issue that devices in the FamShr are only updated after a restart of HA, after restart they are no longer updated. The main device of the saved Apple ID is still updated regularly. I went back to v3.0.4, where everything works perfectly.

Best regards and thank you for your good work.

**Comment by gcobb321** [comment]
Hmmm. I made a number of changes in v3.0.5 setting up and verifying the FamShr devices when iC3 starts. I found that if an error occurred setting up the devices, the retry routines were not getting data from the Apple account but instead just recycling through the data it already had, which was incomplete. The change I made was to get the data again and then continue with the setup process.

The only thing I can think of was you encountered some type of problem where the devices were not verified so they would not be located.

Can you help me figure out what is going on? I would like log files of v3.0.4 and v3.0.5.1 so I can see the differences.

With v3.0.4 running:
1. Start RawData logging (Event Log > Actions > Start RawData logging)
2. Restart HA to initialize the icloud3-0.log file. Let it run for a few minutes until the initial locate is done.
3. Rename the _icloud3-0.log_ file to _icloud3-0-v304.log_ to save it.
4. Reinstall v3.0.5.1 from HACS or the way you normally install updates.
5. Restart HA to create a new icloud3-0.log file. Let it run for a few minutes until the initial locate is done.
3. Rename the _icloud3-0.log_ file to _icloud3-0-v305.log_ to save it.
4. Zip the -v304 and -v305 log files and post theme here or email them to geekstergary@gmail.com

Then reinstall v3.0.4 and restart HA.

That will give me a good comparison between the versions to see why it worked in v3.0.4 and did not in v3.0.5.

Thanks

**Comment by gcobb321** [comment]
The problem is caused by more than one device in the Family Sharing List with the same name. I have reverted to the detection code in v3.0.4 so the v3.0.5.2 .zip file below should take care of it. I also added a warning message and alert to the Event Log The message was there but you wouldn't see it if you did not review the setup info, Also, the duplicate device that was located last will be used when it is encountered.

Unzip the v3.0.5.2 zip file into the icloud3 directory and restart HA.
[icloud3 v3.0.5.2.zip](https://github.com/gcobb321/icloud3_v3/files/15484701/icloud3.v3.0.5.2.zip)

I am going out of town for a week so won't be able to handle any other fixes until I get back. Send me another rawdata log like you did before if you run into anything.

Reinstall v3.0.4 if things don't go well.
